layer-atlas/src/main/java/com/layer/atlas/AtlasAvatar.java

```diff
@@ -453,23 +423,21 @@ public int hashCode() {
 
     private static Diff diff(Set<Identity> oldSet, Set<Identity> newSet) {
         Diff diff = new Diff();
-        for (Identity old : oldSet) {
-            if (newSet.contains(old)) {
-                diff.existing.add(old);
-            } else {
-                diff.removed.add(old);
-            }
-        }
         for (Identity newItem : newSet) {
-            if (!oldSet.contains(newItem)) {
+            if (!oldSet.contains(newItem) || oldSet.contains(newItem)) {
```

Comment:
is this `if` needed? You check whether `oldSet` doesn't contain or contains `newItem`, is there a third state?

Reply:
true that :D